### PR TITLE
DOCS: improving documentation of Maxwell class

### PR DESCRIPTION
**Pull request by anur7**
## Description
Adding examples and improving docstring of Maxwell's methods. 

## Issue linked
#6136 

## Checklist
- [ ] I have tested my changes locally.
- [ ] I have added necessary documentation or updated existing documentation.
- [ ] I have followed the coding style guidelines of this project.
- [ ] I have added appropriate tests (unit, integration, system).
- [ ] I have reviewed my changes before submitting this pull request.
- [ ] I have linked the issue or issues that are solved by the PR if any.
- [ ] I have agreed with the Contributor License Agreement ([CLA](https://developer.ansys.com/form/cla-acceptance)).

### CODE REVIEW

**Comment by Samuelopez-ansys** [comment]
@anur7 Please could you be more specific in the PR Title? It is important because if you are going to change only one method, then I would change the title to "Improve xxxx method documentation" 

And then please add some description.

Thank you!

File: src/ansys/aedt/core/maxwell.py [hunk 1]
```diff
@@ -111,7 +119,8 @@ def change_inductance_computation(self, compute_transient_inductance=True, incre
             The default is ``True``.
         incremental_matrix : bool, optional
             Whether to set the inductance calculation to ``Incremental`` if
-            ``compute_transient_inductance=True``. The default is ``False``.
+            ``compute_transient_inductance=True``. The default is ``False``,
+            in which case the "Apparent" inductance calculation is enabled.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py:123]:
```suggestion
            ``compute_transient_inductance=True``. The default is ``False``,
            in which case the "Apparent".
```

File: src/ansys/aedt/core/maxwell.py [hunk 2]
```diff
@@ -1082,7 +1158,7 @@ def assign_floating(self, assignment, charge_value=0, name=None):
 
         Examples
         --------
-        Assign a floating excitation for a Maxwell 2d Electrostatic design
+        Assign a floating excitation for a Maxwell 2d Electrostatic design.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: 2D or 3-D

File: src/ansys/aedt/core/maxwell.py [hunk 3]
```diff
@@ -1091,13 +1167,16 @@ def assign_floating(self, assignment, charge_value=0, name=None):
         >>> floating = m2d.assign_floating(assignment=rect, charge_value=3, name="floating_test")
         >>> m2d.release_desktop(True, True)
 
-        Assign a floating excitation for a Maxwell 3d Electrostatic design providing an object
+        Assign a floating excitation for a Maxwell 3d Electrostatic design providing an object.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: The same

File: src/ansys/aedt/core/maxwell.py [hunk 4]
```diff
@@ -1177,6 +1257,16 @@ def assign_winding(
         References
         ----------
         >>> oModule.AssignWindingGroup
+
+        Examples
+        --------
+        Assign a winding for a Maxwell 2d Transient design.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: The same. Could you please check the whole doc?

File: src/ansys/aedt/core/maxwell.py [hunk 5]
```diff
@@ -529,6 +561,7 @@ def setup_ctrlprog(
     def eddy_effects_on(self, assignment, enable_eddy_effects=True, enable_displacement_current=True):
         """Assign eddy effects on a list of objects.
 
+        Available only for Eddy Current and Transient solvers.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: The 3D AC Magnetic A-Phi solver does not support eddy effects off.

File: src/ansys/aedt/core/maxwell.py [hunk 6]
```diff
@@ -1951,6 +2153,35 @@ def export_element_based_harmonic_force(
...
+        >>>                                         name="Coil_terminal")
+        >>> region = m3d.modeler.create_region()
+        >>> m3d.assign_winding(assignment=terminal.name, current="5*cos(2*PI*50*time)")
+        >>> m3d.enable_harmonic_force(assignment=coil.name, force_type=2, calculate_force="Harmonic")
+        >>> setup = m3d.create_setup()
+        >>> setup.props["StopTime"] = "2/50s"
+        >>> setup.props["TimeStep"] = "1/500s"
+        >>> m3d.analyze(setup=setup.name, use_auto_settings=False)
+        >>> m3d.export_element_based_harmonic_force()
+        >>> m3d.release_desktop(True, True)
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py:2191]: How about mentioning about the what the output of this method?

File: src/ansys/aedt/core/maxwell.py [hunk 7]
```diff
@@ -1020,7 +1085,7 @@ def assign_voltage(self, assignment, amplitude=1, name=None):
     def assign_voltage_drop(self, assignment, amplitude=1, swap_direction=False, name=None):
         """Assign a voltage drop across a list of faces to a specific value.
 
-        The voltage drop applies only to sheet objects.
+        The voltage drop applies only to sheet objects. Available only for Magnetostatic 3D.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: It is available...

File: src/ansys/aedt/core/maxwell.py [hunk 8]
```diff
@@ -1042,6 +1108,16 @@ def assign_voltage_drop(self, assignment, amplitude=1, swap_direction=False, nam
...
+
+        Examples
+        --------
+        Create a cylinder in Maxwell 3D and assign voltage drop to one of its face.
+
+        >>> from ansys.aedt.core import Maxwell3d
+        >>> m3d = Maxwell3d(solution_type="Magnetostatic")
+        >>> cylinder = m3d.modeler.create_cylinder(origin=[0, 0, 0], radius=5, height=15, orientation="Z")
+        >>> m3d.assign_voltage_drop(assignment=cylinder.top_face_z, amplitude="1V", name="Volt", swap_direction=False)
+        >>> m3d.release_desktop(True, True)
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py:1121]: Is the excitation set in such a way the model would work?

File: src/ansys/aedt/core/maxwell.py [hunk 9]
```diff
@@ -1262,6 +1365,16 @@ def assign_coil(self, assignment, conductors_number=1, polarity="Positive", name
         References
         ----------
         >>> oModule.AssignCoil
+
+        Examples
+        --------
+        Assign a coil to an object for a Maxwell 2d Transient design.
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: Coil terminal

File: src/ansys/aedt/core/maxwell.py [hunk 10]
```diff
@@ -1934,6 +2133,9 @@ def export_element_based_harmonic_force(
     ):
         """Export an element-based harmonic force data to a .csv file.
 
+        To apply this method, first it is needed to enable element based harmonic force,
```
Comment (nunesanderson) [inline src/ansys/aedt/core/maxwell.py]: Does the user need to run first and then run the script to apply this method?

File: src/ansys/aedt/core/maxwell.py (hunk 5 above)
Comment (anur7) [reply src/ansys/aedt/core/maxwell.py]: @nunesanderson, @gmalinve AC Magnetic A-Phi solver is still beta that's why I did not mention it.

File: src/ansys/aedt/core/maxwell.py (hunk 8 above)
Comment (anur7) [reply src/ansys/aedt/core/maxwell.py:1121]: @nunesanderson here just showing in a generic way how you can use the method to assign the voltage drop excitation

File: src/ansys/aedt/core/maxwell.py (hunk 10 above)
Comment (anur7) [reply src/ansys/aedt/core/maxwell.py]: @nunesanderson here I meant before using this method it is needed to enable element based harmonic force and simulate the model. If you look at the example below, first I enabled the element based harmonic force, run the model, and after used the method to export element based harmonic force.

File: src/ansys/aedt/core/maxwell.py (hunk 6 above)
Comment (anur7) [reply src/ansys/aedt/core/maxwell.py:2191]: I think it is clear reading the above description. As written the output is a .csv file containing element based harmonic force data

File: src/ansys/aedt/core/maxwell.py [hunk 11]
```diff
@@ -111,7 +121,8 @@ def change_inductance_computation(self, compute_transient_inductance=True, incre
             The default is ``True``.
         incremental_matrix : bool, optional
             Whether to set the inductance calculation to ``Incremental`` if
-            ``compute_transient_inductance=True``. The default is ``False``.
+            ``compute_transient_inductance=True``. The default is ``False``,
+            in which case the "Apparent" inductance calculation is enabled.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
            ``compute_transient_inductance=True``. The default is ``False``,
            in which case the ``Apparent`` inductance calculation is enabled.
```

File: src/ansys/aedt/core/maxwell.py [hunk 12]
```diff
@@ -121,6 +132,15 @@ def change_inductance_computation(self, compute_transient_inductance=True, incre
         References
         ----------
         >>> oDesign.SetDesignSettings
+
+        Examples
+        --------
+        Set "Incremental" as inductance calculation.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        Set ``Incremental`` as inductance calculation.
```

File: src/ansys/aedt/core/maxwell.py [hunk 13]
```diff
@@ -529,6 +561,7 @@ def setup_ctrlprog(
     def eddy_effects_on(self, assignment, enable_eddy_effects=True, enable_displacement_current=True):
         """Assign eddy effects on a list of objects.
 
+        Available only for AC Magnetic (Eddy Current) and Transient solvers.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py:565]:
```suggestion
        Available only for Eddy Current/AC Magnetic and Transient solvers.
```
@anur7 try to be consistent in the docstring. Sometimes you refer to solver as EddyCurrent/AC Magnetic and Sometimes you put them within parenthesis.

File: src/ansys/aedt/core/maxwell.py [hunk 14]
```diff
@@ -923,6 +976,17 @@ def assign_rotate_motion(
         References
         ----------
         >>> oModule.AssignBand
+
+        Examples
+        --------
+        Assign rotational motion to a band cointaing all moving objects.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        Assign rotational motion to a band containing all moving objects.
```

File: src/ansys/aedt/core/maxwell.py [hunk 15]
```diff
@@ -953,16 +1017,17 @@ def assign_rotate_motion(
 
     @pyaedt_function_handler(face_list="assignment")
     def assign_voltage(self, assignment, amplitude=1, name=None):
-        """Assign a voltage source to a list of faces in Maxwell 3D or a list of objects or edges in Maxwell 2D.
+        """Assign a voltage excitation to a list of faces in Maxwell 3D or a list of objects or edges in Maxwell 2D.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        """Assign a voltage excitation to a list of faces or edges in Maxwell 2D or a list of objects in Maxwell 2D.
```
@anur7 please look at the implementation.

File: src/ansys/aedt/core/maxwell.py [hunk 16]
```diff
@@ -953,16 +1017,17 @@ def assign_rotate_motion(
...
         ----------
         assignment : list
-            List of faces, objects or edges to assign a voltage source to.
+            List of faces, objects or edges to assign a voltage excitation to.
         amplitude : float, optional
             Voltage amplitude in mV. The default is ``1``.
         name : str, optional
-            Name of the boundary. The default is ``None``.
+            Name of the excitation.
+            If not provided, a random name with prefix "Voltage" will be generated.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
            Name of the excitation.
            If not provided, a random name with prefix ``Voltage`` will be generated.
```
@anur7 please use these quotes (````) when mentioning names.

File: src/ansys/aedt/core/maxwell.py [hunk 17]
```diff
@@ -976,21 +1041,21 @@ def assign_voltage(self, assignment, amplitude=1, name=None):
 
         Examples
         --------
-
         Create a region in Maxwell 2D and assign voltage to its edges.
+
         >>> from ansys.aedt.core import Maxwell2d
         >>> m2d = Maxwell2d(version="2025.1", solution_type="ElectrostaticZ")
         >>> region_id = m2d.modeler.create_region(pad_value=[500, 50, 50])
         >>> voltage = m2d.assign_voltage(assignment=region_id.edges, amplitude=0, name="GRD")
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py:1050]: @nunesanderson @anur7 stupid question but what's the point of assigning a voltage excitation to region's edges?

File: src/ansys/aedt/core/maxwell.py [hunk 18]
```diff
@@ -1031,7 +1096,8 @@ def assign_voltage_drop(self, assignment, amplitude=1, swap_direction=False, nam
         swap_direction : bool, optional
             Whether to swap the direction. The default value is ``False``.
         name : str, optional
-            Name of the boundary. The default is ``None``.
+            Name of the excitation.
+            If not provided, a random name with prefix "VoltageDrop" will be generated.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
            Name of the excitation.
            If not provided, a random name with prefix ``VoltageDrop`` will be generated.
```

File: src/ansys/aedt/core/maxwell.py [hunk 19]
```diff
@@ -1166,7 +1245,8 @@ def assign_winding(
         phase : float, optional
             Value of the phase delay in degrees. The default is ``0``.
         name : str, optional
-            Name of the boundary. The default is ``None``.
+            Name of the winding. The default is ``None``,
+            in which case a random name with prefix "Winding" will be generated.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
            in which case a random name with prefix ``Winding`` will be generated.
```

File: src/ansys/aedt/core/maxwell.py [hunk 20]
```diff
@@ -1449,6 +1571,7 @@ def solve_inside(self, name, activate=True):
         """Solve inside to generate a solution inside an object.
 
         With this method, Maxwell will create a mesh inside the object and generate the solution from the mesh.
+        In Maxwell, by default when an object is created "solve inside" is already enabled.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        In Maxwell, by default when an object is created ``solve inside`` is already enabled.
```

File: src/ansys/aedt/core/maxwell.py [hunk 21]
```diff
@@ -1466,6 +1589,16 @@ def solve_inside(self, name, activate=True):
         References
         ----------
         >>> oEditor.ChangeProperty
+
+        Examples
+        --------
+        Disable "solve inside" of an object.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]: @anur7 same here

File: src/ansys/aedt/core/maxwell.py [hunk 22]
```diff
@@ -1623,6 +1794,20 @@ def assign_current_density(
         -------
         bool
             ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+        >>> oModule.AssignSymmetry
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]: @anur7 why AssignSymmetry? is this not the assign_current_density method?

File: src/ansys/aedt/core/maxwell.py [hunk 23]
```diff
@@ -1934,6 +2133,9 @@ def export_element_based_harmonic_force(
     ):
         """Export an element-based harmonic force data to a .csv file.
 
+        To use this method, first it is needed to enable element based harmonic force,
+         and then solve the model.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        This method requires enabling element-based harmonic force before solving the model.
```

File: src/ansys/aedt/core/maxwell.py [hunk 24]
```diff
@@ -2171,6 +2438,9 @@ def export_rl_matrix(
     ):
         """Export R/L matrix after solving.
 
+        This method allows to export in an .txt file Re(Z)/Im(Z), inductive Coupling Coefficient, R/L,
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        This method allows to export in a .txt file Re(Z)/Im(Z), inductive Coupling Coefficient, R/L,
```

File: src/ansys/aedt/core/maxwell.py [hunk 25]
```diff
@@ -2265,6 +2560,9 @@ def export_c_matrix(
     ):
         """Export Capacitance matrix after solving.
 
+        This method allows to export in an .txt file capacitance and capacitance coupling coefficient
+        matrices for Electrostatic solutions.
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py]:
```suggestion
        This method allows to export in a .txt file capacitance and capacitance coupling coefficient
        matrices for Electrostatic solutions.
```

File: src/ansys/aedt/core/maxwell.py [hunk 26]
```diff
@@ -3132,7 +3523,7 @@ def assign_resistive_sheet(
         >>> bound = m3d.assign_resistive_sheet(assignment=resistive_face, resistance="3ohm")
         >>> m3d.solution_type = SOLUTIONS.Maxwell3d.Magnetostatic
         >>> bound = m3d.assign_resistive_sheet(assignment=resistive_face, non_linear=True)
-        >>> m3d.release_desktop()
+        >>> m3d.release_desktop(True, True)
```
Comment (gmalinve) [inline src/ansys/aedt/core/maxwell.py:3531]: @anur7 just as a small comment because I've seen you added True, True where they were missing. If not provided to release_Desktop the default values are already True, True

**Review by gmalinve — CHANGES_REQUESTED** [review]
@anur7 @nunesanderson I trust all examples are running smoothly as I didnìt have time to test them all. I also trust that all sentences "it works only in..." a specific solver are also correct as I didn't check upon every of them.

File: src/ansys/aedt/core/maxwell.py (hunk 13 above)
Comment (nunesanderson) [reply src/ansys/aedt/core/maxwell.py:565]: I would suggest using "AC Magnetic (Eddy Current)"as in the UI.

File: src/ansys/aedt/core/maxwell.py (hunk 17 above)
Comment (nunesanderson) [reply src/ansys/aedt/core/maxwell.py:1050]: Physically speaking, it's only valid if the region is large enough.

**Review by nunesanderson — DISMISSED** [review]

File: src/ansys/aedt/core/maxwell.py (hunk 6 above)
Comment (nunesanderson) [reply src/ansys/aedt/core/maxwell.py:2191]: Could you please check the format of the file?

File: src/ansys/aedt/core/maxwell.py [hunk 27]
```diff
@@ -1485,17 +1618,33 @@ def analyze_from_zero(self):
...
+        >>> setup.props["TimeStep"] = "1/500s"
+        >>> m2d.analyze_from_zero()
+        >>> m2d.release_desktop(True, True)
         """
         if self.solution_type != SolutionsMaxwell3D.Transient:
             raise AEDTRuntimeError("This methods work only with Maxwell Transient Analysis.")
 
         self.oanalysis.ResetSetupToTimeZero(self._setup)
-        self.analyze()
+        self.analyze(use_auto_settings=False)
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/maxwell.py]: Btw, why is that change performed ? I would not expect such a change in a documentation PR

Comment (anur7) [reply src/ansys/aedt/core/maxwell.py]: @SMoraisAnsys when I was testing the example to include in "analyze_from_zero()" doc, I found that "analyze_from_zero()" was not working because of a bug was located in the changed line. I know that it is not related to documentation, but I thought it is a small change so I did it.

Comment (SMoraisAnsys) [reply src/ansys/aedt/core/maxwell.py]: Ok, just wanted to be sure that this wasn't a missing line introduced while working.
Pinging @nunesanderson @gmalinve for visibility :)

**Review by nunesanderson — APPROVED** [review]

**Comment by nunesanderson** [comment]
@gmalinve I think this PR is ready to be merged. The example with harmonic force is working properly.
Would you mind checking whether Abdun can merge?
There seems to be an issue with the codecov.
Thank you!
@anur7

**Review by gmalinve — DISMISSED** [review]

File: src/ansys/aedt/core/maxwell.py (hunk 27 above)
Comment (gmalinve) [reply src/ansys/aedt/core/maxwell.py]: @anur7 the method fails for the baloon boundary condition because the solver is not able to parallelize it. In order to make it work for vector potential and baloon you should disable the default settings and set the core to 1.
Tests also need to be added.

Comment (gmalinve) [reply src/ansys/aedt/core/maxwell.py]: @anur7 @nunesanderson you can decide to roll back the changes in the code in this PR but fix the docstring by using the vector potential and open an issue to fix and test this method.

Comment (anur7) [reply src/ansys/aedt/core/maxwell.py]: @gmalinve @nunesanderson  I rolled back the changes in this PR adding vector potential. I'll open an issue to fix and test this method.

**Review by gmalinve — APPROVED** [review]

File: src/ansys/aedt/core/maxwell.py (hunk 27 above)
Comment (anur7) [reply src/ansys/aedt/core/maxwell.py]: @gmalinve @nunesanderson I opened the issue #6404
